app/services/course/assessment/submission/update_service.rb

```diff
@@ -153,10 +153,14 @@ def update_submission # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComple
           end
         end
       end
+
+      monitoring_service&.stop! if finalise?
```

Comment:
I dont think we should block/rollback submission finalising if for some reasons we cant stop the monitoring service. We can let it expires after 24 hours or separately just let users stop it..